example/src/screens.tsx

```diff
@@ -76,7 +76,7 @@ export type RootDrawerParamList = {
 };
 
 export type RootStackParamList = ExampleScreensParamList & {
-  Home: NavigatorScreenParams<RootDrawerParamList> | undefined;
+  Home: NavigatorScreenParams<RootDrawerParamList>;
```

Comment:
Why did you change this? Please revert to how it was

Reply:
Could you please help me with how linking options should be defined here if the `Home` route is  `NavigationScreenParams<RootDrawerParamList> | undefined`. I'm getting this typecheck error when it can be undefined:
```
Object literal may only specify known properties, and 'screens' does not exist in type 'Omit<PathConfig<{}>, "initialRouteName" | "screens">'
```
on this row https://github.com/react-navigation/react-navigation/blob/d0abdee67f5db8cf39112af535846ffededfb21d/example/src/index.tsx#L69

I have no experience from using `NavigatorScreenParams` in a union with `undefined` and really no idea how it should be handled by the lib. Would really appreciate some guidance. Thanks!

Reply:
We are omitting `screens` from the type if it's not extending `NavigatorScreenParams` here: https://github.com/react-navigation/react-navigation/blob/d0abdee67f5db8cf39112af535846ffededfb21d/packages/core/src/types.tsx#L1012C37-L1012C46 

Since this PR removes `NonNullable` from that line (because `NonNullable` broke recursive infer) I think that makes it impossible for a route to be union of `NavigatorScreenParams` and `undefined`?

Reply:
If it makes it impossible to have `undefined` then it'd be a breaking change. The change needs to support `| undefined` to be merged

Reply:
Ok, but if a route has is defined as `NavigatorScreenParams<> | undefined` should you be able to config linking options for the nested navigator as in the example?

Reply:
It wouldn't make sense to support `| undefined` if it only works with some of the things and not others.